goblin/driver/connection.py

```diff
@@ -35,17 +32,14 @@ def __init__(self, response_queue, loop, conn, username, password,
             raise StopAsyncIteration
 
     async def fetch_data(self):
-        if not self._response_queue.empty():
-            message = self._response_queue.get_nowait()
-        else:
-            self._loop.create_task(self._conn.get_data())
-            message = await self._response_queue.get()
-        return message
+        if self._done:
+            return None
 
```

Comment:
The other option was to raise an exception, but I figured this would be okay since `None` already signifies the end of the request.